### PR TITLE
Remove "post" from block titles

### LINKED ISSUE #53459: Remove "Post" prefix for all template tag blocks
Originally suggested in https://github.com/WordPress/gutenberg/issues/26856 and recently discussed in https://github.com/WordPress/gutenberg/issues/53427.

The "Post" prefix in block titles like "Post Date", "Post Author", "Post Content" suggest the block is only applicable to `post`s which can be confusing when working on a template for a different post type, e.g. pages, products, etc.

"Post Title" was recently [changed](https://github.com/WordPress/gutenberg/pull/50147) to just "Title", creating inconsistency in the block name convention. Now might be a good time to update related block titles to match.

### PR DESCRIPTION
**Pull request by richtabor**
## What?

Removes the "Post" prefix on block titles. 
Examples, changes "Post Featured Image" to "Featured Image". Closes https://github.com/WordPress/gutenberg/issues/53459. 

## Why?
- Was suggested a few times, notably in https://github.com/WordPress/gutenberg/issues/53459.
- The "Post" prefix is confusing if editing a page, or any other custom post type template.
- There are already a few blocks without the "Post" prefix. 

## Visual 

| Before  | After |
| ------------- | ------------- |
|CleanShot 2023-08-09 at 16 17 49|CleanShot 2023-08-09 at 16 17 23|

### CODE REVIEW

File: packages/block-library/src/post-date/block.json
Comment (fabiankaegy) [inline packages/block-library/src/post-date/block.json:5]: I think just date is confusing as it could just as well be the current date. Maybe this should just be called Published Date

File: packages/block-library/src/post-content/block.json
Comment (fabiankaegy) [inline packages/block-library/src/post-content/block.json:5]: I know I'm thinking about it too much as a programmer. But just content seems to vague to me. 

The post content has become such an abstracted term for the content field.

**Review by fabiankaegy — COMMENTED** [review]
On a general note I really like the simplicity this brings. 

However in a few cases I'm not sure the name as proposed here is super clear. 

I left inline comments for those.

File: packages/block-library/src/post-terms/block.json
Comment (fabiankaegy) [inline packages/block-library/src/post-terms/block.json]: Just germs on its own also feels a little confusing to me. Which terms? Ones attached to the current post or terms in general?

File: packages/block-library/src/post-date/block.json
Comment (richtabor) [reply packages/block-library/src/post-date/block.json:5]: Hm, maybe so. 🤔

File: packages/block-library/src/post-content/block.json
Comment (richtabor) [reply packages/block-library/src/post-content/block.json:5]: > The post content has become such an abstracted term for the content field.

For developers yes, but does "Post content" work when you're editing the images on a portfolio item?

File: packages/block-library/src/post-terms/block.json
Comment (richtabor) [reply packages/block-library/src/post-terms/block.json]: Is the post-terms block available, as-is? Or is it a wrapper for the actual terms blocks?

File: packages/block-library/src/post-content/block.json
Comment (carolinan) [reply packages/block-library/src/post-content/block.json:5]: It is even strange to call it post content when you are building a *page* template

File: packages/block-library/src/post-date/block.json
Comment (carolinan) [reply packages/block-library/src/post-date/block.json:5]: Calling it "published date" would help separate it from the "modified date" variation.

File: packages/block-library/src/post-date/block.json
Comment (carolinan) [inline packages/block-library/src/post-date/block.json]: Maybe for a follow-up, but what about the word "post"  in the descriptions?

**Comment by jasmussen** [comment]
Big fan of this change. I can't today test the PR so I'll let others validate the code and give a green light. But appreciate the simplification 👍 👍

File: packages/block-library/src/post-terms/block.json
Comment (carolinan) [reply packages/block-library/src/post-terms/block.json]: Using the inserter or typing. `/terms` lets us choose between categories and tags.
But in the Site Editor Styles sidebar, we can select the Post Terms" block.

File: packages/block-library/src/post-content/block.json
Comment (jameskoster) [reply packages/block-library/src/post-content/block.json:5]: If we kept the prefix it'd need to be something generic that works in any context... "Entry Content"? I'm not sure that makes it any less vague though. On balance it seems okay to try "Content" – we can always circle back if it proves troublesome.

File: packages/block-library/src/post-terms/block.json
Comment (jameskoster) [reply packages/block-library/src/post-terms/block.json]: I don't think Post Terms should be visible in the UI (it doesn't appear in inserters as noted). It's essentially a base for more contextual blocks referencing taxonomies etc. For reference if you insert it manually it doesn't do anything until you select a variation:

Screenshot 2023-08-10 at 11 37 10

This makes it a more dev oriented block, and since developers are likely to intuit "Post" to mean _post type_ rather than _blog post_, perhaps it's okay to leave the title?

File: packages/block-library/src/post-date/block.json
Comment (jameskoster) [reply packages/block-library/src/post-date/block.json]: Good point. Would making that less explicit work?

> Display the publish date for an entry such as a post or page.

**Comment by skorasaurus** [comment]
I'm a fan of this change: 
although there are blocks (next post, previous post, post list) where removing the post prefix is confusing; Rich kept the post prefix where it makes sense to; 
 but I suggest doing a quick user test to a few casual site editors before merging this?

File: packages/block-library/src/post-date/block.json
Comment (mtias) [reply packages/block-library/src/post-date/block.json:5]: It's actually a bit weird for me to see modified date as a variation in the inserter. I'd expect it to be a single date block and for modified to be a property there.

File: packages/block-library/src/post-terms/block.json
Comment (richtabor) [reply packages/block-library/src/post-terms/block.json]: Adding "Post Terms" back as the title.

File: packages/block-library/src/post-date/block.json
Comment (richtabor) [reply packages/block-library/src/post-date/block.json]: > Display the publish date for an entry such as a post or page.

Done.

**Comment by richtabor** [comment]
> although there are blocks (next post, previous post, post list) where removing the post prefix is confusing; Rich kept the post prefix where it makes sense to;

Yes, exactly. :)

**Review by jasmussen — APPROVED** [review]
I'm happy to try this one.

**Comment by annezazu** [comment]
Heads up to @femkreations for docs for 6.4!

**Pull request merged**
